### PR TITLE
Update howtos so all diffs can be applied and all pass example tests

### PR DESCRIPTION
**Pull request by danielsuo**
Some notes:
- Just updated tests to match whatever was in `train.train` 
- Had to change loss/accuracy thresholds slightly for the ensembling example

Hopefully nothing wrong or controversial.

Once this is in, we can verify that #248 is working (confirmed that #248 with this PR works locally).

### CODE REVIEW

File: howtos/diffs/ensembling.diff [hunk 1]
```diff
@@ -94,26 +94,43 @@ index 51d2fde..0ca3365 100644
...
+     optimizer, train_metrics = train_epoch(
          optimizer, train_ds, batch_size, epoch, input_rng)
      loss, accuracy = eval_model(optimizer.target, test_ds)
 -    logging.info('eval epoch: %d, loss: %.4f, accuracy: %.2f',
 +    # `loss` and `accuracy` are now 1D arrays of length `jax.device_count()`
 +    logging.info('eval epoch: %d, loss: %s, accuracy: %s',
                   epoch, loss, accuracy * 100)
-   return optimizer
- 
+     summary_writer.scalar('train_loss', train_metrics['loss'], epoch)
```
Comment (avital) [inline howtos/diffs/ensembling.diff:104]: I know this isn't a problem in this PR, but I have confused myself now -- what is the "train loss" scalar for an ensemble training?

I'm just putting it here to share my confusing but it shouldn't block merging this PR.

File: howtos/diffs/polyak-averaging.diff [hunk 2]
```diff
@@ -61,6 +61,21 @@ index 51d2fde..eab7983 100644
...
+index fb02770..4cfec2a 100644
+--- a/examples/mnist/train_test.py
++++ b/examples/mnist/train_test.py
+@@ -34,8 +34,8 @@ class TrainTest(absltest.TestCase):
+     input_rng = onp.random.RandomState(0)
+     model = train.create_model(random.PRNGKey(0))
+     optimizer = train.create_optimizer(model, 0.1, 0.9)
+-    optimizer, train_metrics = train.train_epoch(optimizer, train_ds, 128, 0,
+-                                                 input_rng)
++    optimizer, train_metrics = train.train_epoch(optimizer, model.params, train_ds,
```
Comment (avital) [inline howtos/diffs/polyak-averaging.diff:77]: This seems wrong? `train_epoch` does not take a params 2nd argument?

Comment (avital) [reply howtos/diffs/polyak-averaging.diff:77]: (Do tests pass on this HOWTO? I'm surprised)

Comment (danielsuo) [reply howtos/diffs/polyak-averaging.diff:77]: I think this diff of diff might be confusing to read; the signature of train_epoch that this diff of diff references is with respect to the original example, but if you look at the original diff file itself (ie polyak-averaging.diff), the signature of train_epoch changes.

(Ah shoot, on mobile and copied the line not the link to the line):

+def train_epoch(optimizer, params_ema, train_ds, batch_size, epoch, rng):

Comment (danielsuo) [reply howtos/diffs/polyak-averaging.diff:77]: Link: https://github.com/google/flax/blob/dea96db3fa779b25c32ab8c40f4b55a7d26fec93/howtos/diffs/polyak-averaging.diff#L32

Comment (avital) [reply howtos/diffs/polyak-averaging.diff:77]: Ah, I see. This is why the tests weren't passing originally.
